lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -630,12 +611,14 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
     -- doing such a thing is considered bonkers and this is not a behavior we
     -- ought to support.
 
-    let extraInputScripts = case toInpScriptsM of
-            Just toInpScripts ->
-                map (toInpScripts . (view #address) . snd)
-                $ extraInputs <> extraCollateral'
-            Nothing ->
-                []
+    let extraInputScripts =
+            case utxoAssumptions of
+                AllKeyPaymentCredentials -> []
+                AllByronKeyPaymentCredentials -> []
+                AllScriptPaymentCredentialsFrom _template toInpScripts ->
+                    toInpScripts . view #address . snd <$>
+                        extraInputs <> extraCollateral'
```

Comment:
Proposal: save one level of indent and one line:

```suggestion
    let extraInputScripts = case utxoAssumptions of
            AllKeyPaymentCredentials -> []
            AllByronKeyPaymentCredentials -> []
            AllScriptPaymentCredentialsFrom _template toInpScripts ->
                toInpScripts . view #address . snd <$>
                    extraInputs <> extraCollateral'
```

Reply:
Yeah, I'm on the bring about this one: 

1. on one hand, as you point out, leaving a case on prev. line allows to save some spaces.

1. on the other - it reads bit awkward as `case` is shifted to the right and alternatives are shifted to the left, and when reading it you need to go zigzag. You'd never format case expression like this if you'd do it in isolation from its context.

I tend to give slightly more preference to the second point in situations where there is enough horizontal space not to hang `case` off the cliff, where saving horizontal space doesn't make the rest less wrapped. Also, it feels weird to first chose 4 space indentation instead of 2 space, and then care about horizontal space 😝